py/runtime.c

```diff
@@ -319,6 +319,15 @@ mp_obj_t mp_unary_op(mp_unary_op_t op, mp_obj_t arg) {
             // if arg==mp_const_none.
             return mp_const_true;
         }
+        #if MICROPY_PY_BUILTINS_FLOAT
+        if (op == MP_UNARY_OP_FLOAT_MAYBE
+            #if MICROPY_PY_BUILTINS_COMPLEX
+            || op == MP_UNARY_OP_COMPLEX_MAYBE
+            #endif
+            ) {
+            return MP_OBJ_NULL;
+        }
+        #endif
```

Comment:
@dpgeorge do you think it's worth checking / splitting for float & complex separately like this, bit harder to read but technically accurate?

Reply:
This is more accurate and actually documents more explicitly what is going on.  So let's leave it as you have it.